File: app/src/generic/java/com/owncloud/android/utils/PushUtils.java
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2019 Chris Narkiewicz <hello@ezaquarii.com>
 * SPDX-FileCopyrightText: 2017 Mario Danic <mario@lovelyhq.com>
 * SPDX-License-Identifier: AGPL-3.0-or-later OR GPL-2.0-only
 */
package com.owncloud.android.utils;

import android.content.Context;

import android.accounts.Account;
import android.text.TextUtils;

import com.google.gson.Gson;
import com.nextcloud.client.account.UserAccountManager;
import com.nextcloud.client.preferences.AppPreferencesImpl;
import com.owncloud.android.MainApp;
import com.owncloud.android.datamodel.ArbitraryDataProviderImpl;
import com.owncloud.android.datamodel.PushConfigurationState;
import com.owncloud.android.datamodel.SignatureVerification;
import com.nextcloud.unifiedpush.UnifiedPush;
import com.owncloud.android.ui.activity.DrawerActivity;

import java.security.Key;

public final class PushUtils {
    public static final String KEY_PUSH = "push";

    private PushUtils() {
    }

    public static void updateRegistrationsWithServer(
        final DrawerActivity activity,
        final UserAccountManager accountManager,
        final String pushToken) {
        for (Account account : accountManager.getAccounts()) {
            String providerValue = new ArbitraryDataProviderImpl(MainApp.getAppContext()).getValue(account.name, KEY_PUSH);
            PushConfigurationState accountPushData = new Gson().fromJson(providerValue, PushConfigurationState.class);

            if ((accountPushData != null) && accountPushData.isShouldBeDeleted()) {
                // unregister push notifications
                UnifiedPush.Companion.unregisterForPushMessaging(MainApp.getAppContext(), account.name);
            } else {
                // else, (re-)register for push notifications
                if (activity != null) {
                    UnifiedPush.Companion.registerForPushMessaging(
                        activity,
                        account.name,
                        ((accountPushData == null) || TextUtils.isEmpty(accountPushData.getPushToken())));
                }
            }
        }
    }

    public static void reinitKeys(UserAccountManager accountManager) {
        AppPreferencesImpl.fromContext(MainApp.getAppContext()).setKeysReInitEnabled();
    }

    public static Key readKeyFromFile(boolean readPublicKey) {
        return null;
    }

    public static SignatureVerification verifySignature(
        final Context context,
        final UserAccountManager accountManager,
        final byte[] signatureBytes,
        final byte[] subjectBytes) {
        return null;
    }

}

File: src/generic/java/com/owncloud/android/utils/PushUtils.kt
/**
 * Nextcloud Android client application
 *
 * @author Mario Danic
 * @author Chris Narkiewicz
 * Copyright (C) 2017 Mario Danic
 * Copyright (C) 2019 Chris Narkiewicz
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
 */
package com.owncloud.android.utils

import android.content.Context
import com.nextcloud.client.account.UserAccountManager
import com.nextcloud.client.preferences.AppPreferencesImpl
import com.owncloud.android.MainApp
import com.owncloud.android.datamodel.SignatureVerification
import java.security.Key

object PushUtils {
    const val KEY_PUSH = "push"
    @JvmStatic
    fun pushRegistrationToServer(accountManager: UserAccountManager?, pushToken: String?) {
        // do nothing
    }

    @JvmStatic
    fun reinitKeys(accountManager: UserAccountManager?) {
        val context = MainApp.getAppContext()
        AppPreferencesImpl.fromContext(context).setKeysReInitEnabled()
    }

    fun readKeyFromFile(readPublicKey: Boolean): Key? {
        return null
    }

    fun verifySignature(
        context: Context?,
        accountManager: UserAccountManager?,
        signatureBytes: ByteArray?,
        subjectBytes: ByteArray?
    ): SignatureVerification? {
        return null
    }
}
